CBIIT/crdc-datahub-ui
# CRDCDH-922 Cross Submission Validation

### Overview

This PR introduces new functionality to perform validation across submissions within the same study. 

> [!NOTE]
> The results of cross validation will NOT appear in the Validation Results tab yet. See CRDCDH-988 for status, but this does not impact the implementation.

> [!Tip]
> **Manual testing notes**:
> Create 2-3 new data submissions (the intention does not matter). I would recommend updating the submissions you created to modify the `studyAbbreviation` to something custom (e.g. "test-922"). This will prevent cross-validation conflicts with all of the existing data submissions and has no impact to the application.
> 
> Scenario II Steps:
> 1. Submit two of the submissions, and try to release one of them. The "Release" button should be disabled, but the "Cross Validate" button should be visible
> 2. Perform cross validation, manually update the `crossSubmissionStatus` to `Passed` (if it did not pass)
> 3. You should be able to Release now
>
> Scenario I Steps:
> 1. One remaining submission should be submitted, and one should still be `In Progress`
> 2. Try to Release the submitted one, you should see the dialog with special text
> 3. You can Release 
>
> Scenario 0 Steps (No cross validation or warning):
> 1. You have one remaining submission "In Progress"
> 2. You can "Admin Submit" it and also "Release" without any CRDCDH-922 features being involved 

### Change Details (Specifics)

- Add new "Cross Validate" button
  - Add test coverage for new button based on requirements
- Add new restrictions to the "Release" action
  - Add test coverage for release utils
- Add test coverage for ValidationControls core functionality. This will help prevent issues like:
  - #348
  - #347 
  - #341
- Add test coverage for MetadataUpload component
- Update QC Tab to refresh the table when cross-submission validation updates
- Update Submission type definition and queries for new properties: `otherSubmissions` and `crossSubmissionStatus`

### Related Ticket(s)

CRDCDH-922
